sky/execution.py

```diff
@@ -176,6 +176,15 @@ def _execute(
         existing_handle = global_user_state.get_handle_from_cluster_name(
             cluster_name)
         cluster_exists = existing_handle is not None
+    if cluster_exists:
+        assert len(task.resources) == 1
```

Comment:
Come to think of it, one can indeed do `Task().set_resources({})` or pass None there. Should we fix this in `set_resources()`?

Reply:
Maybe we can prevent such a case in the `task.set_resources` method. I think it's not relevant to this PR though.

Reply:
Yes, please add a TODO?

Reply:
Added.